packages/build-tools/src/common/easBuildInternal.ts

```diff
@@ -44,29 +44,42 @@ export async function runEasBuildInternalAsync<TJob extends BuildJob>({
     autoSubmitArgs.push('--auto-submit');
   }
 
-  const result = await spawn(
-    cmd,
-    [
-      ...args,
-      'build:internal',
-      '--platform',
-      job.platform,
-      '--profile',
-      buildProfile,
-      ...autoSubmitArgs,
-    ],
-    {
-      cwd,
-      env: {
-        ...env,
-        EXPO_TOKEN: nullthrows(job.secrets, 'Secrets must be defined for non-custom builds')
-          .robotAccessToken,
-        ...extraEnv,
-      },
-      logger,
-      mode: PipeMode.STDERR_ONLY_AS_STDOUT,
-    }
-  );
+  let result;
+  try {
+    result = await spawn(
+      cmd,
+      [
+        ...args,
+        'build:internal',
+        '--platform',
+        job.platform,
+        '--profile',
+        buildProfile,
+        ...autoSubmitArgs,
+      ],
+      {
+        cwd,
+        env: {
+          ...env,
+          EXPO_TOKEN: nullthrows(job.secrets, 'Secrets must be defined for non-custom builds')
+            .robotAccessToken,
+          ...extraEnv,
+        },
+        logger,
+        mode: PipeMode.STDERR_ONLY_AS_STDOUT,
+      }
+    );
+  } catch (err: any) {
+    // Logging of build:internal is weird. We set mode = STDERR_ONLY_AS_STDOUT,
+    // because we don't want to pipe stdout to to the logger as it will contain
+    // the job object with its secrets. However, if the build fails, stderr alone
+    // is often not enough to debug the issue, so we also log stdout in those cases.
+    // It will look awkward (first stderr from pipe, then stdout from here), but
+    // it's better than nothing.
+    logger.error(`${err.stdout}`);
```

Comment:
I assume that printing job object with it's secrets is the last operation performed by this command, right? If so if there is any error there shouldn't be any secrets in stdout?

Reply:
It's not last per se, but it's close to last. After we print the job we may also start a submission (which theoretically may fail).

Maybe we should revert this after we find the issue with the orchestrator?

Reply:
Yeah, we can if it contains any sensitive data and there is a risk of leaking it 🤔